### PR TITLE
Specifications by default unselected, additional links to "Select All"

### LINKED ISSUE #177: Update MP Starter to uncheck the 8 specs
Based on the google group [discussion](https://groups.google.com/forum/#!topic/microprofile/LGJqv-UG3oU) , the general consensus is:
By default, uncheck the 8 specs. Add a button/link to say "Select All".

If possible, try to add "Unselect All", but check to see whether it will clatter the website. If it does, leave this out.

### PR DESCRIPTION
**Pull request by rdebusscher**
and "Unselect All". Fixes #177.

### CODE REVIEW

**Comment by Karm** [comment]
Live here for everyone to try: https://test-start.microprofile.io/

**Comment by rdebusscher** [comment]
I propose to make the font a bit smaller (however, no issue on my screen which is 1680 x 1050)

**Comment by Karm** [comment]
@rdebusscher How about this?

See https://test-start.microprofile.io/

Recording:

output

All in all, no string opinion as long as there is no linebreak....

**Comment by Karm** [comment]
Without @rstjames, we slowly but surely butcher the design :rofl:

**Comment by rstjames** [comment]
@Karm 
I can suggest this solution. It is more of a standard on the UI side of things. It keeps things short and everyone has used this function once or twice before. Similar to gmail.

It is not designed just a screenshot of the idea. Let me know what you think.

Screen Shot 2019-10-10 at 10 47 13 AM

**Comment by Karm** [comment]
Hello @Emily-Jiang, 
 * there was a version implemented by @rdebusscher,
 * there was a version modified by me (what is live on the test site) and
 * there was a version suggested by @rstjames 

Since Ryan is the designer, we will follow what he suggested and we modify this PR accordingly soon.

**Comment by Karm** [comment]
@Emily-Jiang @rdebusscher The version suggested by @rstjames is live on https://test-start.microprofile.io/ covered in a PR #232 

WDYT?

**Comment by rdebusscher** [comment]
We could at least discussed this first before you created all those versions ...

**Comment by Karm** [comment]
@rdebusscher I was toying with it to get more familiar with UI where I have 0 experience. That is why I was poking at it.

**Comment by Karm** [comment]
"Select / Clear All" without a checkbox is the conclusion from the call Oct 14th.

File: src/main/webapp/index.xhtml [hunk 1]
```diff
@@ -120,12 +120,20 @@
                             </p:selectOneMenu>
                         </p:panelGrid>
                         <h:panelGroup>
-                            <p:outputLabel value="Examples for specifications" for="microprofileSpecs"/>
+                            <p:outputLabel value="Examples for specifications" for="microprofileSpecs"/> <h:outputText value="&#160;" />
+                            <p:commandLink value="Select All" actionListener="#{generatorDataBean.selectAll}" update="microprofileSpecs"
+                                           ajax="true" process="@this"
+                                           rendered="#{not empty generatorDataBean.engineData.mpVersion}"/>
+                            <h:outputText value="&#160;" /><h:outputText value="&#160;" /><h:outputText value="&#160;" />
+                            <p:commandLink value="Unselect All" actionListener="#{generatorDataBean.unselectAll}" update="microprofileSpecs"
```
Comment (Emily-Jiang) [inline src/main/webapp/index.xhtml]: Based on this week's discussion, Unselect All will be Clear All.

**Comment by rdebusscher** [comment]
Remark: Due to a wrong commit, it is best that PR #231 is merged before this one to avoid merge conflicts.

**Comment by Emily-Jiang** [comment]
I tested the changes on the test server but the functionality did not work. Do you know why?

**Comment by Emily-Jiang** [comment]
ah. I think the gitter message explained the mastery.

**Comment by rdebusscher** [comment]
After I merged #231, I updated this PR to remove the merge conflicts.  @Karm @Emily-Jiang

**Comment by Karm** [comment]
@rdebusscher Let's merge this one too then....

**Pull request merged**